core/chains/evm/client/node.go

```diff
@@ -452,18 +467,50 @@ func (n *node) HeaderByNumber(ctx context.Context, number *big.Int) (header *typ
 	return
 }
 
-func (n *node) SendTransaction(ctx context.Context, tx *types.Transaction) error {
+func (n *node) SendTransaction(ctx context.Context, tx *types.Transaction) (err error) {
+	lggr := n.newRqLggr(switching(n)).With("tx", tx)
+
+	var rpcDomain string
+	defer func(start time.Time) {
+		duration := time.Since(start)
+
+		// In the event where the remote node is down, we will return before
+		// we have a chance to set rpcDomain to something useful.
+		if rpcDomain == "" {
```

Comment:
Does getting the domain depend on the remote node being up? I thought we'd be able to get it anyways

Reply:
I guess we can just grab the domain first based on whether `http` or `ws` is available, but I'm not sure how useful the prom metric would be if we don't actually end up sending the request to the remote node, now that I think about it.

Maybe something like

```golang
var rpcDomain string
if n.http != nil {
	rpcDomain = n.http.uri.Host
} else {
	rpcDomain = n.ws.uri.Host
}
defer func(...) {
// ...
}(...)
```

Reply:
If the `err != nil` clause is true here:

```golang
ctx, cancel, err := n.makeLiveQueryCtx(ctx)
if err != nil {
	return err
}
defer cancel()
```

we'd get super low numbers for `duration`, though maybe we can just filter on `successful = true` in Grafana?

Reply:
Right I think we'd want to filter on success = true usually. Timing on success=false though is still interesting sometimes due to timeouts

Reply:
Ah I think in that case we might want to make the bucket > than the context timeout.

Reply:
Or I guess we can leave it as is, it'll just be in the > 8 second bucket.